early-dev/app.R

```diff
@@ -1,141 +1,175 @@
-  teal.data::col_relabel(
-    TRTDURD = "Treatment Duration in Days",
-    DTHFL = "Death Flag",
-    DCSREAS = "Reason for Study Discontinuation",
-    EOSSTT = "End of Study Status"
-  ) %>%
-  droplevels()
-teal.data::col_labels(ADSL)[c(names(adsl_labels))] <- adsl_labels
-
-ADAE <- synthetic_cdisc_data("latest")$adae
-
-# derive common flags for AEs
-adae_labels <- teal.data::col_labels(ADAE, fill = FALSE)
-ADAE <- ADAE %>%
-  mutate_at(c("AESOC", "AEBODSYS", "AEHLT", "AEDECOD", "AETERM", "AELLT"), as.character) %>%
-  mutate(
-    RELFL = ifelse(AEREL == "Y", "Y", "N"),
-    CTC35FL = ifelse(AETOXGR %in% c("3", "4", "5"), "Y", "N"),
-    SERFL = ifelse(AESER == "Y", "Y", "N"),
-    RELSERFL = ifelse(AEREL == "Y" & AESER == "Y", "Y", "N"),
-    AEREL1 = (AEREL == "Y" & ACTARM == "A: Drug X"),
-    AEREL2 = (AEREL == "Y" & ACTARM == "B: Placebo"),
-    ASTDT = as.Date(ASTDTM),
-    AENDT = as.Date(AENDTM)
-  ) %>%
-  teal.data::col_relabel(
-    RELFL = "Related AE",
-    CTC35FL = "Grade >=3 AE",
-    SERFL = "Serious AE",
-    RELSERFL = "Related Serious AE",
-    AEREL1 = "AE related to A: Drug X",
-    AEREL2 = "AE related to B: Placebo",
-    ASTDT = "Analysis Start Date",
-    AENDT = "Analysis End Date",
-    AESOC = "Primary System Organ Class",
-    AEBODSYS = "Body System or Organ Class",
-    AEHLT = "High Level Term",
-    AEDECOD = "Dictionary-Derived Term",
-    AETERM = "Reported Term for the Adverse Event",
-    AELLT = "Lowest Level Term"
-  )
+  ADTR <- synthetic_cdisc_data("latest")$adtr
+
+  # process ADTR
+  adtr_labels <- teal.data::col_labels(ADTR, fill = FALSE)
+  ADTR <- ADTR %>%
+    mutate(
+      PCHG = ifelse(AVISIT == "BASELINE", 0, PCHG),
+      CHG = ifelse(AVISIT == "BASELINE", 0, CHG),
+      AVAL = ifelse(AVISIT == "BASELINE", BASE, AVAL),
+      AVALC = ifelse(AVISIT == "BASELINE", as.character(BASE), AVALC)
+    ) %>%
+    filter(AVISIT != "SCREENING")
+  teal.data::col_labels(ADTR) <- adtr_labels
+
+  ADTRWF <- ADTR %>%
+    filter(AVISIT != "BASELINE")
+  teal.data::col_labels(ADTRWF) <- teal.data::col_labels(ADTR, fill = FALSE)
+
+
+  # process ADRS
+  ADRSSWIM <- synthetic_cdisc_data("latest")$adrs %>%
+    filter(PARAMCD == "OVRINV") %>%
+    arrange(USUBJID)
+
+  ADRS <- synthetic_cdisc_data("latest")$adrs
+  adrs_labels <- teal.data::col_labels(ADRS, fill = FALSE)
+  ADRS <- ADRS %>%
+    filter(PARAMCD %in% c("BESRSPI", "INVET")) %>%
+    mutate(ADT = as.Date(ADTM)) %>%
+    droplevels()
+  teal.data::col_labels(ADRS) <- c(adrs_labels, "Analysis Date")
+
+  ADLB <- synthetic_cdisc_data("latest")$adlb
+
+  # process ADLB
+  ADLB <- ADLB %>%
+    mutate(
+      ADT = as.Date(ADTM),
+      LBSTRESN = as.numeric(LBSTRESC)
+    ) %>%
+    teal.data::col_relabel(
+      ADT = "Analysis Date",
+      LBSTRESN = "Numeric Result/Finding in Standard Units"
+    )
+})
+
+# set datanames
+datanames <- c("ADSL", "ADAE", "ADCM", "ADEX", "ADTR", "ADTRWF", "ADRS", "ADRSSWIM", "ADLB")
+datanames(data) <- datanames
```

Comment:
It would be nice to use something like `datanames <- ls(data@env)`. We don't want the user to play with slots though.

Reply:
Yes, but then you have  `adtr_labels` (class: character) in the list of datasets

Reply:
I mean only do it when applicable, i.e. all existing variables are data sets. I assumed that is the case here as well, I missed `adtr_labels`.

Reply:
We can also do it automatically in the first `eval_code`. We could set  `@datanames <- names(@env)` when `@datanames` is empty.

Reply:
Wouldn't you have to modify `eval_code` to do that?

Reply:
Nah, true. Anyway, setting `join_keys` brings a meaningful information.